Req resp

shparekh00 opened the pull request:
This branch is about changing the master slave terminology to requestor responder.
Jira Issue: https://gem5.atlassian.net/browse/GEM5-9

Commit:
Added ResponsePort class title

shparekh00 committed:
mem: wrote RequestPort in port.hh

Change-Id: I48133eae72279456b168338f93ae05315a04ff23

Commit:
Made master/slave deprecated subclasses and made request and response classes.

shparekh00 committed:
mem: Changing Master - Slave terms

Changed a couple files in src/mem folder to use
RequestPort and RespondPort

Change-Id: I0d9da7ed820d0e9305b8c8ad941bba9654b50973

shparekh00 committed:
mem: Merge branch to origin

Change-Id: I0a68e394aa74a216af71aac66402dee935a97291

shparekh00 committed:
mem: Changed files in src/mem to use Request and Response

Change-Id: I0a68e394aa74a216af71aac66402dee935a97291

eavivi4 committed:
mem: Port.cc, qport.hh, port_proxy MasterSlave change

Change-Id: I637c28e8122fb6ce497b35cc2bf734f451490acc

eavivi4 committed:
mem: port.cc, qport.hh, port_proxy MasterSlave change

Change-Id: I32b4b125d4211d3915168130e84d9d1f17752627

shparekh00 committed:
mem: Merge branch 'shivani/reqresp' into ReqResp

Change-Id: I5516db31274725aa1068090cb0c0fa218be83a2e

shparekh00 committed:
mem: Merge branch 'ReqResp'

merge changes from files in src/mem

Change-Id: Ifa34b9bc46291d3a2142ce9a88707951aee0d02e

eavivi4 committed:
mem: Master-Slave change

Packet_queue, noncoherent_xbar, MemDelay, MemChecker files changed

Change-Id: I9876c5342941becd7e6958b858674f79a2cba7dd

Commit:
mem: Updated files with master/slave terminology

Change-Id: Ie515265dd5a0f1e82761ae46b109ec4706fc47db

Commit:
mem: Updated files with master/slave terminology

Change-Id: Ia023db928284260aaa3ab10c34673a47ebb18f01

eavivi4 committed:
mem: MasterID change

Change-Id: I13acbfac954c9c18b92b5e9fcaf88ba63314a1e9

shparekh00 committed:
mem: Changing MemId to RequestorId

Change-Id: I63a89e9ab6fedefb3aa159e2dca635139840cd70

shparekh00 committed:
mem: Merge branch 'shivani/reqresp' into ReqResp

Changes made to remove MasterId from Shivani's section of files

Change-Id: Ia6c87d58b696bf3d6ea10bf0586e1bc0f5ccec09

shparekh00 committed:
mem: Replaced ExternalSlave and ExternalMaster

Change-Id: If60d22442debe637286bc90661cc6fc0ca49e1ce

shparekh00 committed:
mem: Merge branch 'shivani/reqresp' into ReqResp

changes consist of ExternalSlave and ExternalMaster

Change-Id: I0aa4ec8e8f99eb3b1d75605e1202a7b0f8f6488d

powerjg commented on src/mem/AddrMapper.py:
```diff
@@ -37,19 +37,19 @@
     # one port in each direction
-    master = MasterPort("Master port")
-    slave = SlavePort("Slave port")
+    requestor = RequestPort("Request port")
```
Let's update this documentation while we're here... (sorry for pointing this out a bit late). Something like what's below.

```suggestion
    requestor = RequestPort("Port which forwards the original request to the 'memory side.' The packet being sent as the request will have the new address based on the configured mapping.")
```

LakinS commented on src/mem/noncoherent_xbar.cc:
```diff
@@ -53,34 +53,35 @@
-        QueuedSlavePort* bp = new NoncoherentXBarSlavePort(portName, *this, i);
-        slavePorts.push_back(bp);
+    // create the response ports, once again starting at zero
+    for (int i = 0; i < p->port_response_connection_count; ++i) {
```
I was thinking that this might be confusing. Would variables like this with 'port_response' instead of 'response_port' be better as 'port_responder' ?
-Lakin

powerjg commented on src/mem/AddrMapper.py:
```diff
@@ -37,19 +37,19 @@
-    master = MasterPort("Master port")
-    slave = SlavePort("Slave port")
+    requestor = RequestPort("Request port")
+    responder = ResponsePort("Response port")
```
Maybe we should call this receiver? 

```suggestion
    receiver = ResponsePort("Port that receives requests and translates them before forwarding out of the requestor port.")
```

powerjg commented on src/mem/AddrMapper.py:
```diff
@@ -37,19 +37,19 @@
 from m5.SimObject import SimObject
 
 # An address mapper changes the packet addresses in going from the
```
We should move this comment to be a true python documentation comment:

E.g.,

```
"""An address mapper changes...
"""
```

And it should be under `class AddrMapper`

Finally, it would be good to mention here that you can use the `splice` function to insert this address mapper between two already connected objects.

powerjg reviewed (commented):
A couple of comments to chat about.

eavivi4 committed:
mem: Master/Slave changes

Previous files fixed and
fixed MaterID on
src/dev, src/sim, src/systemc and util

Change-Id: Ie847a179559ce195baa9a643674a3efa8c631aa2

ebrickey committed:
mem: refactoring some masterinfo/id ref and other master refs

Change-Id: Ied50fcfb8d54a739f04190026d8af1badb755529

Commit:
mem: changed terminology for MasterID/masterId

Change-Id: Iafc2fefd547a7609d5063697e84caa45a73c4cc3

ebrickey committed:
mem: merged reqResp

Change-Id: I1ee7b280f24617535f095d54ceb8561700a910d2

Commit:
mem: changed terminology for MasterID/masterId

Change-Id: I35c27d55e0d5b693f2a2b651fb2b323965c8a060

Commit:
mem: terminology for MasterID

Change-Id: Ieb1cd9af4f795e88bdda4f54eabdfc9ead236e43

ebrickey committed:
mem: sc_master_port changed references to sc_request_port

Change-Id: I3ea48185ae87363182282631333a19a0df60d462
